clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java

```diff
@@ -457,9 +474,16 @@ private synchronized void bootstrap() {
       List<Integer> partitions = storeNameToPartitionListMap.get(storeName);
       String versionTopic = version.kafkaTopicName();
       LOGGER.info("Bootstrapping partitions {} for {}", partitions, versionTopic);
-      aggVersionedStorageEngineStats.setStorageEngine(versionTopic, storageService.getStorageEngine(versionTopic));
+      AbstractStorageEngine storageEngine = storageService.getStorageEngine(versionTopic);
+      aggVersionedStorageEngineStats.setStorageEngine(versionTopic, storageEngine);
       StoreBackend storeBackend = getStoreOrThrow(storeName);
-      storeBackend.subscribe(ComplementSet.newSet(partitions), Optional.of(version));
+
+      if (recordTransformer != null) {
+        recordTransformer
+            .onRecovery(storageEngine, storeBackend, blobTransferManager, partitions, Optional.of(version));
+      } else {
+        storeBackend.subscribe(ComplementSet.newSet(partitions), Optional.of(version));
```

Comment:
Why are we 'else'ing here?  It seems we're doing so because we want all steps in onREcovery to come before storeBackend.subscribe is called.  But that would just happen anyway by virtue of .onRecovery being called before subscribe unless theres something I'm missing.

Reply:
The intention behind this structure was that if blob transfer was to work with the record transformer, we would need to create a custom subscribe method that would pass in the record transformer so that it can later be referenced when blob transfer is invoked.

But since blob transfer has been paused, I can go ahead and remove the else and subscribe call within onRecovery.

Reply:
Is this still the case even with the notion that the hash is stored in rocksdb itself and therefore doesn't need to be in the interface?

Reply:
If the hash was stored in RocksDB, then we would actually need subscribe to complete first before we start reading from RocksDB and calling processPut.

Reply:
By moving `onRecovery` to SIT, bootstrap will complete before `onRecovery` is executed.